### PR TITLE
New Validation::(min|max)ByteLength() addition

### LINKED ISSUE #9817: New Validation::maxByteLength() addition
This is a (multiple allowed):

* [ ] bug
* [x] enhancement
* [x] feature-discussion (RFC)

* CakePHP Version: 2.9.3

What's the reason that `Validation::maxLength` is in characters. When storing data in the database maximum lengths (e.g. `VARCHAR(255)`) are always in bytes right?

And are there any objections to adding a `$bytes` option (default `false`)?

### PR DESCRIPTION
**Pull request by tersmitten**
Fixes #9817

### CODE REVIEW

**Comment by markstory** [comment]
Looks decent to me.

File: src/Validation/Validation.php [hunk 1]
```diff
@@ -676,7 +676,7 @@ public static function ip($check, $type = 'both')
     }
 
     /**
-     * Checks whether the length of a string is greater or equal to a minimal length.
+     * Checks whether the length of a string (in character) is greater or equal to a minimal length.
```
Comment (dereuromark) [inline src/Validation/Validation.php]: Should that by (like bytes) plural "characters"?

Comment (tersmitten) [inline src/Validation/Validation.php]: Yes, I'll fix it

**Comment by tersmitten** [comment]
Fixed the typo

**Comment by tersmitten** [comment]
Do I also need to add methods in the `Validator` class?

**Comment by dereuromark** [comment]
Will you add the new method to the documentation?

**Comment by dereuromark** [comment]
Ah, right. That Validator class methods are convenience wrappers.

**Comment by tersmitten** [comment]
So I also add them to `Validator`?

**Comment by tersmitten** [comment]
The docs go [here](https://github.com/cakephp/docs), right?

**Comment by tersmitten** [comment]
https://github.com/cakephp/docs/pull/4619

**Comment by tersmitten** [comment]
@dereuromark 

> Will you add the new method to the documentation?

Only for Cake 2, right? [See](https://book.cakephp.org/3.0/en/core-libraries/validation.html)

**Pull request merged**